src/pybind/mgr/cephadm/module.py

```diff
@@ -1123,6 +1133,7 @@ def _refresh_host_devices(self, host):
             host, len(devices), len(networks)))
         devices = inventory.Devices.from_json(devices)
         self.cache.update_host_devices_networks(host, devices.devices, networks)
+        self.cache.update_osdspec_previews(host)
```

Comment:
Hm. `update_host_devices_networks()` only updates the cache, while `update_osdspec_previews()` is doing calls to remote hosts. Yes, what about moving this to OSDService?

Reply:
The method `update_host_devices_networks` is called in does remote calls however. It's just one less(or one more, however you want to look at it) level of abstraction..

Reply:
ok, maybe we can put it into module.py then?

Reply:
sure